core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -570,6 +572,46 @@ public ServerBuilder maxConnectionAge(Duration maxConnectionAge) {
         return maxConnectionAgeMillis(requireNonNull(maxConnectionAge, "maxConnectionAge").toMillis());
     }
 
+    /**
+     * Sets the connection drain duration in micros for the connection shutdown.
+     * At the beginning of the connection drain server signals the clients that the connection shutdown is
+     * imminent but still accepts in flight requests.
+     * After the connection drain end server stops accepting new requests.
+     * Also, see {@link ServerBuilder#connectionDrainDuration(Duration)}.
+     *
+     * <p>
+     * Note that HTTP/1 doesn't support draining as described here, so for HTTP/1 drain duration
+     * is always {@code 0}.
+     * </p>
+     *
+     * @param durationMicros the drain duration. {@code 0} or negative value disables the drain.
+     */
+    public ServerBuilder connectionDrainDurationMicros(long durationMicros) {
```

Comment:
https://datatracker.ietf.org/doc/html/rfc7540#section-6.8 suggests that drain duration can be as low as round-trip time. Roundtrip inside the datacenter is < 1ms, around few hundred microseconds. If client is connected via localhost (e.g. sidecar process) it's gonna be few microseconds. I chose microsecond precision to give users an ability to configure in those lower ranges if they need to.

Reply:
+1 Make sense. Let's keep it as it is.

Reply:
I'm somewhat concerned about using micros for a long parameter because this is probably the only place we use micros instead of millis. Would it be a bad idea to accept a millis value here? A user can use `Duration` for micros precision.

Reply:
For this builder method - its name contains `Micros` so there should be little room for the confusion. If you think that providing an override with `Millis` is valuable I can add that. But I think there's value in keeping `Micros` override too, to make it explicit that server builder supports that level of precision for connection drain duration.